src/lib/core/error/error-options.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {InjectionToken} from '@angular/core';
+import {NgControl, FormGroupDirective, NgForm} from '@angular/forms';
+
+/** Injection token that can be used to specify the global error options. */
+export const MD_ERROR_GLOBAL_OPTIONS = new InjectionToken<ErrorOptions>('md-error-global-options');
+
+export type ErrorStateMatcher =
+  (control: NgControl, parentFormGroup: FormGroupDirective, parentForm: NgForm) => boolean;
+
+export interface ErrorOptions {
+  errorStateMatcher?: ErrorStateMatcher;
+}
+
+export function defaultErrorStateMatcher(control: NgControl, formGroup: FormGroupDirective,
+    form: NgForm): boolean {
+
+  const isInvalid = control && control.invalid;
+  const isTouched = control && control.touched;
+  const isSubmitted = (formGroup && formGroup.submitted) ||
+      (form && form.submitted);
+
+  return !!(isInvalid && (isTouched || isSubmitted));
+}
+
+export function showOnDirtyErrorStateMatcher(control: NgControl, formGroup: FormGroupDirective,
+    form: NgForm): boolean {
```

Comment:
I agree with the sentiment, but I'm not clear on how it would work. The `Form` interface doesn't have the `submitted` property, which is the main (if not only) reason we need it. I see two solutions

1. Use the union `FormGroupDirective | NgForm`

```ts
export function showOnDirtyErrorStateMatcher(control: FormControl, form: FormGroupDirective | NgForm) {
  const isInvalid = control.invalid;
  const isDirty = control.dirty;
  const isSubmitted = form && form.submitted;

  return !!(isInvalid && (isDirty || isSubmitted));
}
```
2. Just pass `submitted` and don't expose either `FormGroupDirective` or `NgForm`
```ts
export function showOnDirtyErrorStateMatcher(control: FormControl, submitted: boolean) {
  const isInvalid = control.invalid;
  const isDirty = control.dirty;

  return !!(isInvalid && (isDirty || submitted));
}